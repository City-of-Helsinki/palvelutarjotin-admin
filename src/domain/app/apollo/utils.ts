import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
