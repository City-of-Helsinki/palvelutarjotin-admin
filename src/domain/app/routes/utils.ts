import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
