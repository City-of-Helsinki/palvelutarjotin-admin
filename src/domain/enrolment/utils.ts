import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
