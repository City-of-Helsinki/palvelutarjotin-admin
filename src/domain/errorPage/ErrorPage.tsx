import { Button, IconAttention } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router';

import useLocale from '../../hooks/useLocale';
import Container from '../app/layout/Container';
import styles from './ErrorPage.module.scss';

const NotFoundPage: React.FC<{ title?: string; description?: string }> = ({
  title,
  description,
}) => {
  const history = useHistory();
  const locale = useLocale();
  const { t } = useTranslation();

  const goToFrontPage = () => {
    history.push(`/${locale}`);
  };

  return (
    <Container className={styles.container}>
      <div className={styles.iconWrapper}>
        <IconAttention />
      </div>
      <h1>{title || t('errorPage.title')}</h1>
      <p className={styles.description}>
        {description || t('errorPage.description')}
      </p>
      <Button onClick={goToFrontPage} style={{ maxWidth: '300px' }}>
        {t('errorPage.returnToHome')}
      </Button>
    </Container>
  );
};

export default NotFoundPage;
