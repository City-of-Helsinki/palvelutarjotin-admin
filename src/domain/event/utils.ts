import { SUPPORT_LANGUAGES } from '../../constants';
import { EventQuery } from '../../generated/graphql';
import { Language } from '../../types';
import { EVENT_PLACEHOLDER_IMAGES } from './constants';
/**
 * Get event placeholder image url
 * @param {string} id
 * @return {string}
 */
export const getEventPlaceholderImage = (id: string): string => {
  const numbers = id.match(/\d+/g);
  const sum = numbers
    ? numbers.reduce((prev: number, cur: string) => prev + Number(cur), 0)
    : 0;
  const index = sum % 4;

  return EVENT_PLACEHOLDER_IMAGES[index];
};

/**
 * Get first availanle language of an event
 * @param {object} eventData
 * @return {string}
 */
export const getFirstAvailableLanguage = (eventData: EventQuery): Language => {
  if (eventData.event?.name.en) return SUPPORT_LANGUAGES.EN;
  if (eventData.event?.name.fi) return SUPPORT_LANGUAGES.FI;
  if (eventData.event?.name.sv) return SUPPORT_LANGUAGES.SV;

  return SUPPORT_LANGUAGES.FI;
};
