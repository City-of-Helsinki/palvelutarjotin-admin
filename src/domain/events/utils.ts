import { isFuture } from 'date-fns';
import forEach from 'lodash/forEach';

import { EventFieldsFragment } from '../../generated/graphql';

export const hasOccurrences = (event: EventFieldsFragment) => {
  return Boolean(event.pEvent?.occurrences.edges.length);
};

export const hasComingOccurrences = (event: EventFieldsFragment) => {
  let hasComingItems = false;

  forEach(event.pEvent?.occurrences.edges, (edge) => {
    if (edge?.node?.startTime && isFuture(new Date(edge?.node?.startTime))) {
      hasComingItems = true;
      return false;
    }
  });

  return Boolean(hasComingItems);
};
