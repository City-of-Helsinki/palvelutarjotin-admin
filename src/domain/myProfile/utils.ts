import {
  OrganisationNodeFieldsFragment,
  PersonFieldsFragment,
} from '../../generated/graphql';
import { MyProfileFormFields } from './myProfileForm/MyProfileForm';

export const getMyProfilePayload = (values: MyProfileFormFields) => ({
  name: values.name,
  phoneNumber: values.phoneNumber,
  emailAddress: values.emailAddress,
  organisations: values.organisations,
});

export const getSelectedOrganisation = (
  myProfile: PersonFieldsFragment,
  activeOrganisation: string | null
): OrganisationNodeFieldsFragment | null => {
  const organisations =
    myProfile.organisations.edges.map((edge) => ({
      ...(edge?.node as OrganisationNodeFieldsFragment),
    })) || [];

  const organisation = activeOrganisation
    ? organisations.find((item) => item.id === activeOrganisation)
    : null;

  return organisation || organisations[0];
};
