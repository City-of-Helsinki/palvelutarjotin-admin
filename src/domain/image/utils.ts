import { ImageDocument, ImageQuery } from '../../generated/graphql';
import { initializeApolloClient } from '../app/apollo/apolloClient';

export const getImageName = (id: string) => {
  const apolloClient = initializeApolloClient();
  const data = apolloClient.readQuery<ImageQuery>({
    query: ImageDocument,
    variables: { id },
  });
  return data?.image?.name;
};
