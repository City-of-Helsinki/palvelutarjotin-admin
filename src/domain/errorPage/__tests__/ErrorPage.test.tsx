import userEvent from '@testing-library/user-event';
import * as React from 'react';
import { vi } from 'vitest';
// eslint-disable-next-line import/no-named-as-default
import * as Router from 'react-router-dom';

import { render, screen } from '../../../utils/testUtils';
import messages from '../../app/i18n/fi.json';
import ErrorPage from '../ErrorPage';

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return { ...actual };
});
const navigate = vi.fn();
it('matches snapshot', () => {
  const { container } = render(<ErrorPage />);

  expect(container).toMatchSnapshot();
});

it('renders correct texts when props are provided', () => {
  const title = 'Title';
  const description = 'Description';
  render(<ErrorPage description={description} title={title} />);

  expect(screen.queryByRole('heading', { name: title })).toBeVisible();
  expect(screen.queryByText(description)).toBeVisible();
});

it('render correct default texts', async () => {
  vi.spyOn(Router, 'useNavigate').mockImplementation(() => navigate);
  render(<ErrorPage />);

  expect(screen.queryByText(messages.errorPage.description)).toBeVisible();
  expect(screen.queryByText(messages.errorPage.returnToHome)).toBeVisible();
  expect(screen.queryByText(messages.errorPage.title)).toBeVisible();

  await userEvent.click(
    screen.getByRole('button', { name: messages.errorPage.returnToHome })
  );

  expect(navigate).toHaveBeenCalledTimes(1);
  expect(navigate).toHaveBeenCalledWith('/fi', expect.anything());
});
