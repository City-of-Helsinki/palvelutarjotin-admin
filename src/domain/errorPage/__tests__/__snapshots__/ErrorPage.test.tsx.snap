// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="container default container"
  >
    <div
      class="content"
    >
      <div
        class="iconWrapper"
      >
        <svg
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 6v6.5h2V18H9v-1.5h2v-5H9V10h4zm-1.187-4a1.312 1.312 0 110 2.625 1.312 1.312 0 010-2.625z"
              fill="currentColor"
            />
          </g>
        </svg>
      </div>
      <h1>
        Palvelussa ilmeni virhe
      </h1>
      <p
        class="description"
      >
        Pahoittelemme, mutta palvelussa ilmeni virhe. Voit yrittää uudestaan, mutta jos tilanne toistuu, kerro siitä meille alalaidan palautelinkin kautta.
      </p>
      <button
        class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
        style="max-width: 300px;"
        type="button"
      >
        <span
          class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
        >
          Palaa etusivulle
        </span>
      </button>
    </div>
  </div>
</div>
`;
